src/v/kafka/server/client_quota_translator.h

```diff
@@ -65,6 +65,11 @@ enum class client_quota_type {
     partition_mutation_quota
 };
 
+static constexpr std::array all_client_quota_types = {
```

Comment:
nitpick: this could be in the in the implementation file.

In a header file, prefer `inline` over static for constants.
In a cpp file, prefer anonymous namespace over static.

Reply:
Thanks, yeah I think I'd prefer to keep it in the header to have it next to the enum variants to make sure that it also gets updated. But yeah, `inline` would probably be better.

Reply:
>But yeah, inline would probably be better.

@pgellert i think you forgot to change that to inline

Reply:
To be honest, I forgot about it since it was a nitpick, but thanks for the reminder: https://github.com/redpanda-data/redpanda/pull/20153